lingua_franca/lang/parse_en.py

```diff
@@ -341,6 +341,25 @@ def _extract_whole_number_with_text_en(tokens, short_scale, ordinals):
         prev_word = tokens[idx - 1].word if idx > 0 else ""
         next_word = tokens[idx + 1].word if idx + 1 < len(tokens) else ""
 
+        if is_numeric(word[:-2]) and \
+                (word.endswith("st") or word.endswith("nd") or word.endswith("rd") or word.endswith("th")):
+
+            # explicit ordinals, 1st, 2nd, 3rd, 4th.... Nth
+            if word.endswith("st") and word[:-2].endswith("1"):
+                word = word[:-2]
+            elif word.endswith("nd") and word[:-2].endswith("2"):
+                word = word[:-2]
+            elif word.endswith("rd") and word[:-2].endswith("3"):
+                word = word[:-2]
+            else:
+                word = word[:-2]
```

Comment:
I'm a little confused here, though it's possible I'm just tired. These conditions look like they always end the same, no? `word = word[:-2]` in all four cases.

I'm also wondering whether `if word[:-2].endswith('<char>')` could be `if word[-3] is <char>`

Reply:
indeed, thats a leftover from a previous approach, guess it was me who was tired!